sample-code/spring-app/src/main/java/com/sap/ai/sdk/app/controllers/OrchestrationController.java

```diff
@@ -142,19 +155,29 @@ Object outputFiltering(
       @Nullable @RequestParam(value = "format", required = false) final String format,
       @Nonnull @PathVariable("policy") final AzureFilterThreshold policy) {
 
-    final OrchestrationChatResponse response;
+    final var response = service.outputFiltering(policy);
+
+    final String content;
     try {
-      response = service.outputFiltering(policy);
-    } catch (OrchestrationClientException e) {
-      final var msg = "Failed to obtain a response as the content was flagged by output filter.";
-      log.debug(msg, e);
-      return ResponseEntity.internalServerError().body(msg);
+      content = response.getContent();
+    } catch (OrchestrationOutputFilterException e) {
+      final var msg =
+          new StringBuilder(
+              "Failed to obtain a response as the content was flagged by output filter.");
+
+      Optional.ofNullable(e.getAzureContentSafetyOutput())
+          .map(AzureContentSafetyOutput::getHate)
+          .filter(rating -> rating.compareTo(policy.getAzureThreshold()) > 0)
+          .ifPresent(rating -> msg.append("Hate score %d ".formatted(rating.getValue())));
```

Comment:
```suggestion
      if(e.getAzureContentSafetyOutput() != null){
        msg.append(" Violence score is %d and exceeded filter value %d".formatted(e.getAzureContentSafetyOutput().getViolence().getValue(), policy.getAzureThreshold().getValue()));
      }
```
Your code doesn't work, please also change the input filter